resources/ruby/guess-gems.rb

```diff
@@ -0,0 +1,60 @@
+require 'parser/current'
+require 'json'
+
+# For now, we will only guess gems that come preinstalled in Polygott.
+# The mapping between what you require and the gem name is not always 1-to-1,
+# so we will need some sort of mapping like what we do with PyPi if we want
+# to do complete ruby gem guessing.
+$allowed_gems = [
+  "sinatra",
+  "stripe"
+]
+
+def guess_gems(code, guesses)
+  root = Parser::CurrentRuby.parse(code)
+  traverse_node(root, guesses)
+end
+
+def traverse_node(node, guesses)
+  if node.class != Parser::AST::Node
+    return
+  end
+  
+  # Looking for any s(:send, nil, :require, s(:str, "gem"))
+  if node.type == :send && node.children[1] == :require
+    req_node = node.children[2]
+    if req_node.class == Parser::AST::Node && req_node.type == :str
+      process_require(req_node.children.first, guesses)
+      return
+    end
+  end
+
+  node.children.each { |child|
+    traverse_node(child, guesses)
+  }
+end
+
+def process_require(req_str, guesses)
+  if req_str.empty?
+    return
+  end
+
+  # Skip absolute or relative requires
+  if req_str.start_with?('/') || req_str.start_with?('.')
+    return
+  end
+
+  # Handle requires like 'sinatra' or 'sinatra/content_for'
+  gem = req_str.split('/')[0]
+  if $allowed_gems.include?(gem)
+    guesses[gem] = true
+  end
+end
```

Comment:
Is this too broad? I'm wondering if another import with the "sinatra/" prefix might not necessarily be part of the "sinatra" gem, based on what you wrote in your comment about require names not mapping 1:1 to gem names.

Reply:
Yeah, I'm somewhat torn on this one. For example, this script uses the `parser` gem which I require via `require 'parser/current'`. There's also `sinatra/base` which comes from the `sinatra` gem, but on the flip-side there iss `sinatra/namespace` which comes from `sinatra-contrib`.

Reply:
Actually, I'll just make the script look for `sinatra` and `sinatra/base`, but ignore any other `sinatra/<things>`

Reply:
Cool, being more conservative here seems like the safe move 👍